database/migrations/2020_12_21_210105_fix_bad_ldap_server_url_for_v5.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class FixBadLdapServerUrlForV5 extends Migration
+{
+    /**
+     * Under v4 and previous versions of Snipe-IT, we permitted users to incorrectly specify LDAP URL's in their settings, and Snipe-IT 
+     * would silently permit that.
+     * 
+     * v5's LDAP system is not so lenient, and requires either ldap:// or ldaps:// in front of the server's URL. This migration tries
+     * to find misconfigured LDAP URL's and prepend 'ldap://' to them. (That's what we assumed if we *didn't* see ldaps://)
+     */
+    
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        // UPDATE settings SET ldap_server = CONCAT('ldap://',ldap_server) WHERE ldap_server NOT LIKE 'ldap://%' AND ldap_server NOT LIKE 'ldaps://%'
+        $settings = App\Models\Setting::where("ldap_server","not like","ldap://%")->where("ldap_server","not like","ldaps://%");
```

Comment:
Should that be `->where()` or `->orWhere()`?

Reply:
It definitely needs to be `->where()` - we **do** want the implied `AND` clause there - 

it needs to be `NOT LIKE` `ldap://%` but needs to _also_ be `NOT LIKE` `ldaps://%` - if either one of those were to be false - in otherwords, if it did start with ldap:// or it started with ldaps:// - then we don't want to change the settings.

Reply:
Ah, yeah, it was the negative that was throwing me, sorry!